src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -10,18 +10,95 @@
  */
 
 #include "vulnerabilityScannerFacade.hpp"
+#include "archiveHelper.hpp"
 #include "flatbuffers/include/syscollector_deltas_generated.h"
 #include "flatbuffers/include/syscollector_synchronization_generated.h"
 #include "loggerHelper.h"
+#include "timeHelper.h"
 #include "vulnerabilityScanner.hpp"
 #include "wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp"
+#include "xzHelper.hpp"
 
 constexpr auto VULNERABILITY_SCANNER_TEMPLATE = "queue/indexer/vd_states_template.json";
 constexpr auto DEFAULT_QUEUE_PATH = "queue/sockets/queue";
 constexpr auto REPORTS_QUEUE_PATH = "queue/vd/reports";
 constexpr auto REPORTS_BULK_SIZE {1};
 constexpr auto REPORTS_THREADS_NUMBER {1};
 constexpr auto MICROSEC_FACTOR {1000000};
+constexpr auto VD_STATE_QUEUE_PATH = "queue/vd/state_track";
+constexpr auto VD_KEYSTORE_PATH = "queue/keystore";
+constexpr auto VD_DATABASE_PATH {"queue/vd"};
+
+bool VulnerabilityScannerFacade::decompressDatabase(const std::string& databaseVersion)
+{
+    bool ret = false;
+
+    // Check database version and presence. It will attempt to decompress the database
+    // if the version doesn't not match, the state_track does not have information
+    // or there's no VD DB.
+    if ((databaseVersion.compare(__ossec_version) != 0 || databaseVersion.empty() ||
+         !std::filesystem::exists(VD_DATABASE_PATH)) &&
+        std::filesystem::exists(TMP_DIR))
+    {
+        // Search in path for a compressed database file.
+        std::regex regexPattern(VD_COMPRESSED_FILE_PATTERN);
+        std::string compressedFilePath;
+        size_t compressedFilesNumber = 0;
+        for (const auto& item : std::filesystem::directory_iterator(TMP_DIR))
+        {
+            if (std::regex_match(item.path().filename().string(), regexPattern))
+            {
+                compressedFilePath = item.path().string();
+                ++compressedFilesNumber;
+            }
+            if (compressedFilesNumber > 1)
+            {
+                throw std::runtime_error("Multiple database compressed files detected");
+            }
+        }
+
+        // Check for XZ downloaded file .
+        if (compressedFilePath.empty())
+        {
+            throw std::runtime_error("Missing database compressed file. Check DOWNLOAD_CONTENT option.");
+        }
+
+        logInfo(WM_VULNSCAN_LOGTAG, "Starting database file decompression.");
+        logDebug2(WM_VULNSCAN_LOGTAG, "Starting XZ file decompression.");
+
+        std::string decompressedFilePath = compressedFilePath;
+        Utils::replaceFirst(decompressedFilePath, ".xz", "");
+
+        // Decompress XF file format.
+        Utils::XzHelper(std::filesystem::path(compressedFilePath), std::filesystem::path(decompressedFilePath))
+            .decompress();
+
+        // Extract queue/vd and queue/vd_updater
```

Comment:
Missing `queue/vd_updater`?

Reply:
It's extracting both

Reply:
The comment was a bit measling, because in the code we append `VD_DATABASE_PATH` (queue/vd) to the `extractOnly` vector